diff --git a/bench/bench.cpp b/bench/bench.cpp
index 47a50c4..de27fbb 100755
--- a/bench/bench.cpp
+++ b/bench/bench.cpp
@@ -71,6 +71,7 @@ int main(int argc, char **argv, char **env) {
 	main_time++;
 	top->arbclk_i = !top->arbclk_i;
 	if (main_time%5 == 0) top->clk = !top->clk;
+	if (main_time%100000 == 0) cout<<"Partial sum = "<<hex<<top->sum<<"\n";
     }
 
     cout<<"Final sum = "<<hex<<top->sum<<"\n";
